### PR TITLE
Expand user relative paths

### PR DESCRIPTION
**Pull request by jjv360**
On my Mac the $PATH has `~/.npm-global/bin` in it, so any executables installed by `npm install -g` were not being detected...

Oddly enough the builtin `which` command on Mac also couldn't detect them, yet I could actually run those executables from the command line by just specifying the name, I don't know why that is...

### CODE REVIEW

**Comment by isaacs** [comment]
`which` does do tilde-expansion, but the shell does.  When you type `~/.npm-global/bin/foo` as a command, the shell translates that to `/Users/jjv360/.npm-global/bin/foo` or whatever.  But it _doesn't_ do tilde expansion for the PATH environment variable.  So if you have `/Users/jjv360/.npm-global/bin` in your PATH, it'll find them.  But if you don't, then it won't.

The which builtin only can do `which ~/.npm-global/bin/foo` because the shell translates `~` before calling `which`.

Long story short, probably not gonna add tilde expansion to this library.  It's more complicated than you might expect.
